### PR TITLE
Host Plans

### PR DESCRIPTION
**Pull request by znarf**
Implement: https://github.com/opencollective/opencollective/issues/2662

### CODE REVIEW

**Comment by kewitz** [comment]
Rebased on top of master.

**Comment by znarf** [comment]
Rebased.

**Comment by znarf** [comment]
Rebased again.

File: migrations/20200114123654-host-plans.js [hunk 1]
```diff
@@ -2,27 +2,41 @@
 
 module.exports = {
   up: async (queryInterface, Sequelize) => {
-    await queryInterface.addColumn('Collectives', 'isHostAccount', {
-      type: Sequelize.BOOLEAN,
-      defaultValue: false,
-      allowNull: false,
+    queryInterface.describeTable('Collectives').then(async tableDefinition => {
```
Comment (kewitz) [inline migrations/20200114123654-host-plans.js]: await

File: migrations/20200114123654-host-plans.js [hunk 2]
```diff
@@ -2,27 +2,41 @@
...
-    });
-    await queryInterface.addColumn('Collectives', 'plan', {
-      type: Sequelize.STRING,
-      allowNull: true,
-    });
-    await queryInterface.addColumn('CollectiveHistories', 'plan', {
-      type: Sequelize.STRING,
-      allowNull: true,
+
+    queryInterface.describeTable('CollectiveHistories').then(async tableDefinition => {
```
Comment (kewitz) [inline migrations/20200114123654-host-plans.js]: await

File: server/models/Collective.js [hunk 3]
```diff
@@ -2407,7 +2407,8 @@ export default function(Sequelize, DataTypes) {
       });
       const plan = (tier && tier.data) || plans[this.plan];
       if (plan) {
-        return { name: this.plan, hostedCollectives, addedFunds, ...plan };
+        const extraPlanData = get(this.data, 'plan', {});
+        return { name: this.plan, hostedCollectives, addedFunds, ...plan, ...extraPlanData };
```
Comment (kewitz) [inline server/models/Collective.js]: 👍

**Comment by znarf** [comment]
Rebased.

**Comment by znarf** [comment]
@kewitz  Test are warning or failing:

- cron/monthly/verify-plans.js
  - Unhandled rejection SequelizeForeignKeyConstraintError: insert or update on table "Expenses" violates foreign key constraint "Expenses_CollectiveId_fkey"`
- server/graphql/v1/collective
  - edits
    - apply to host: Only USER or ORGANIZATION can become Host.

Can you look at that?

**Pull request merged**